Tella/Scenes/Uwazi/Utils/UwaziEntityParser.swift

```diff
@@ -8,77 +8,241 @@
+        entityInstance?.server = appModel.tellaData?.getUwaziServer(serverId: template.serverId)
+        entityInstance?.collectedTemplate = appModel.tellaData?.getUwaziTemplateById(id: template.id)
+        entityInstance?.files = uwaziEntityInstanceFile
+        self.entityInstance = entityInstance
+    }
+    
+    func putAnswers() {
+        
+        let metadata = self.entityInstance?.metadata
 
-        } ?? []
-        entryPrompts.append(contentsOf: entryPromptyProperties)
+        let vaultFilesID = self.entityInstance?.files.compactMap{$0.vaultFileInstanceId} ?? []
+        let vaultFiles = appModel.vaultFilesManager?.getVaultFiles(ids: vaultFilesID) ?? []
+        
+        self.entityInstance?.documents = Set(vaultFiles.filter({$0.mimeType?.isPDF ?? false}))
+        self.entityInstance?.attachments = Set(vaultFiles.filter({!($0.mimeType?.isPDF ?? true)}))
+        
+        for entryPrompt in entryPrompts {
+            let propertyName = entryPrompt.name
+            let value = metadata?[propertyName ?? ""]
+            switch  entryPrompt.type {
+                
+            case .dataTypeText where propertyName == UwaziEntityMetadataKeys.title:
+                guard let entryPrompt = entryPrompt as? UwaziTextEntryPrompt else { continue }
+                entryPrompt.value = self.entityInstance?.title ?? ""
+                
+            case .dataTypeText, .dataTypeNumeric, .dataTypeMarkdown, .dataTypeDate:
+                guard let entryPrompt = entryPrompt as? UwaziTextEntryPrompt else { continue }
+                let valueDict = value as? [[String:Any]]
+                guard let decoded = try? valueDict?.first?.decode(UwaziValue<String>.self) else { continue }
+                entryPrompt.value = decoded.value
+                
+            case .dataTypeSelect, .dataTypeMultiSelect:
+                let uwaziString = value as? [[String:Any]]
+                guard let entryPrompt = entryPrompt as? UwaziSelectEntryPrompt else { continue }
+                guard let decoded =  uwaziString?.compactMap({ try? $0.decode(UwaziValue<String>.self)})  else { continue }
+                entryPrompt.value = decoded.compactMap({$0.value})
+                
+            case .dataTypeMultiFiles:
+                guard let entryPrompt = entryPrompt as? UwaziFilesEntryPrompt else { continue }
+                entryPrompt.value = Set(self.entityInstance?.attachments ?? [])
+                
+            case .dataTypeMultiPDFFiles:
+                guard let entryPrompt = entryPrompt as? UwaziFilesEntryPrompt else { continue }
+                entryPrompt.value = Set(self.entityInstance?.documents ?? [])
+            case .dataRelationship:
+                guard let entryPrompt = entryPrompt as? UwaziRelationshipEntryPrompt else { continue }
+                let uwaziString = value as? [[String:Any]]
+                guard let decoded = uwaziString?.compactMap({ try? $0.decode(EntityRelationshipItem.self)}) else { continue }
```

Comment:
Could you use `UwaziValue`  
I think it's better to save only the `id` and it will be 
```
 guard let decoded =  uwaziString?.compactMap({ try? $0.decode(UwaziValue<String>.self)})  else { continue }
 entryPrompt.value = decoded.compactMap({$0.value})

```
as I remember we can only send the `value` without the `label`, but let's try